.circleci/config.yml

```diff
@@ -0,0 +1,47 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+
+# Orbs are reusable packages of CircleCI configuration that you may share across projects, enabling you to create encapsulated, parameterized commands, jobs, and executors that can be used across multiple projects.
+# See: https://circleci.com/docs/2.0/orb-intro/
+orbs:
+  # The python orb contains a set of prepackaged CircleCI configuration you can use repeatedly in your configuration files
+  # Orb commands and jobs help you with common scripting around a language/tool
+  # so you dont have to copy and paste it everywhere.
+  # See the orb documentation here: https://circleci.com/developer/orbs/orb/circleci/python
+  python: circleci/python@2.0.3
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
+jobs:
+  build-docs: # This is the name of the job, feel free to change it to better match what you're trying to do!
+    # These next lines defines a Docker executors: https://circleci.com/docs/2.0/executor-types/
+    # You can specify an image from Dockerhub or use one of the convenience images from CircleCI's Developer Hub
+    # A list of available CircleCI Docker convenience images are available here: https://circleci.com/developer/images/image/cimg/python
+    # The executor is the environment in which the steps below will be executed - below will use a python 3.10.2 container
+    # Change the version below to your required version of python
+    docker:
+      - image: cimg/python:3.10.2
+    # Checkout the code as the first step. This is a dedicated CircleCI step.
+    # The python orb's install-packages step will install the dependencies from a Pipfile via Pipenv by default.
+    # Here we're making sure we use just use the system-wide pip. By default it uses the project root's requirements.txt.
+    # Then run your tests!
+    # CircleCI will report the results back to your VCS provider.
+    steps:
+      - checkout
+      - run:
+          name: Build docs
+          command: |
+            poetry install
+            python -m pip install sphinx pydata-sphinx-theme
+            cd docs; make html
```

Comment:
I think this can be simplified with poetry. Poetry has the concept of [dependency groups](https://python-poetry.org/docs/master/managing-dependencies/#dependency-groups) meaning we can install the doc dependencies by running ``poetry install --with doc``. I think we will be required to change the makefile to work with poetry if we do go down that route though.

Reply:
I am trying but even with the right dependencies in the docs group of the pyproject.toml file, when I do "poetry install --with docs" nothing is installed. To be fair none of the other groups work for me either and "poetry install --with test" or "poetry install --with bench" also don't do anything. I'm really not familiar with poetry so I wonder if I'm missing a step somewhere...

Reply:
The argument should be `docs`, not `doc` (note the extra `s`). It might be that you have installed _all_ the dependencies already in your local environment, which is why it's not installing anything else.

Reply:
Yeah, I figured the doc vs docs thing looking at https://github.com/numpy/numpy-financial/blob/165663904ff7d670495099fb69ae690470d483b7/pyproject.toml#L49

Still, it doesn't work:

```
➜  numpy-financial git:(setup-circle) poetry install --with docs
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: numpy-financial (1.1.0.dev0)
➜  numpy-financial git:(setup-circle) cd doc
➜  doc git:(setup-circle) make html
/bin/sh: linha 1: sphinx-build: comando não encontrado  # "command not found"
make: *** [Makefile:20: html] Error 127
```